libs/sim/simulator.rs

```diff
@@ -201,6 +209,14 @@ impl<'a> Simulator {
                 }
             }
         }
+        if self.record_network_data {
+            log::info!(
+                "Network use in tick {0} is {1}",
+                tick,
+                data_over_network_size
+            );
+        }
         log::info!("################# TICK {0} END #################", tick);
+        return data_over_network_size;
```

Comment:
I do not like this. tick has a specific purpose and this is tacking on stuff. I would keep all this metrics collection separate. You can compute everything you need in an independent csv